Core/StateManager.cs

```diff
@@ -1,90 +1,55 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using Core.Importers;
-using Core.TimeSeries;
 using Newtonsoft.Json;
 
 namespace Core
 {
 	public class StateManager
 	{
-		private static Dictionary<string, TransactionsImporter> importers = new Dictionary<string, TransactionsImporter>
+		private static Dictionary<string, ITransactionsImporter> importers = new Dictionary<string, ITransactionsImporter>
 		{
 			["pb"] = new PrivatTransactionsImporter(),
+			["usb"] = new UkrSibTransactionsImporter()
 		};
 
-		public static void LoadState(string stateJson)
+		public static string SaveCategoriesToJson()
 		{
-			if (string.IsNullOrWhiteSpace(stateJson))
+			return JsonConvert.SerializeObject(State.Instance.Categories);
+		}
+
+		public static void LoadCategories(string categoriesJson)
+		{
+			if (string.IsNullOrWhiteSpace(categoriesJson))
 			{
 				return;
 			}
 
-			State.Instance = JsonConvert.DeserializeObject<State>(stateJson);
-		}
-
-		public static string SaveToJson()
-		{
-			return JsonConvert.SerializeObject(State.Instance);
+			var categories = JsonConvert.DeserializeObject<ICollection<Category>>(categoriesJson);
+			State.Instance = new State(categories, State.Instance.Transactions);
 		}
 
 		public static void LoadTransactions(IEnumerable<(string key, Stream stream)> files)
 		{
-			var transactions = new List<Transaction>();
+			var newTransactions = new List<Transaction>();
 			foreach (var file in files)
 			{
-				transactions.AddRange(importers[file.key].Load(file.stream));
+				newTransactions.AddRange(importers[file.key].Load(file.stream));
 			}
 
-			foreach (var t in transactions)
-			{
-				State.Instance.Transactions.Add(t);
-			}
-		}
-		
-		public static SmoothedTimeSeries GetSmoothedTimeSeries(string category, double smoothingRatio)
-		{
-			var filteredTransactions = State.Instance.Transactions.Where(State.Instance.CategoryFilters[category]).ToList();
-			return new SmoothedTimeSeries(filteredTransactions, smoothingRatio);
-		}
+			var newCategories = newTransactions.Select(t => t.Category).Where(c => c is { } && State.Instance.Categories.All(sc => sc.Name != c))
+				.Distinct().Select(c => new Category(c, new Rule[] { new Rule(".*", "*", ".*", c) }, 1, 10000)).ToList();
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=5924375)